libraries/AP_Math/quaternion.cpp

```diff
@@ -118,6 +118,271 @@ void Quaternion::from_rotation_matrix(const Matrix3f &m)
     }
 }
 
+// create a quaternion from a given rotation
+void Quaternion::from_rotation(enum Rotation rotation)
+{
+    // the constants below can be calculated using the following formula:
+    //     Matrix3f m_from_rot;
+    //     m_from_rot.from_rotation(rotation);
+    //     Quaternion q_from_m;
+    //     from_rotation_matrix(m_from_rot);
+
+    switch (rotation) {
+    case ROTATION_NONE:
```

Comment:
need test suite in examples/rotations for this

Reply:
test suite added, thanks!